presto-hive/src/main/java/com/facebook/presto/hive/metastore/CachingHiveMetastore.java

```diff
@@ -317,6 +323,28 @@ private PartitionStatistics loadPartitionColumnStatistics(HivePartitionName part
         return result.build();
     }
 
+    @Override
+    public void updateTableStatistics(String databaseName, String tableName, Function<PartitionStatistics, PartitionStatistics> update)
+    {
+        try {
+            delegate.updateTableStatistics(databaseName, tableName, update);
+        }
+        finally {
+            tableStatisticsCache.invalidate(HiveTableName.table(databaseName, tableName));
+        }
+    }
+
+    @Override
+    public void updatePartitionStatistics(String databaseName, String tableName, String partitionName, Function<PartitionStatistics, PartitionStatistics> update)
+    {
+        try {
+            delegate.updatePartitionStatistics(databaseName, tableName, partitionName, update);
+        }
+        finally {
+            partitionStatisticsCache.invalidate(HivePartitionName.partition(databaseName, tableName, partitionName));
```

Comment:
should not we invalidate tableStatistics as well?

Reply:
No. If the table is partitioned, the table statistics are always empty.